tree/ntuplebrowse/v7/src/RNTupleBrowser.cxx

```diff
@@ -0,0 +1,161 @@
+/// \file ROOT/RNTupleBrowser.cxx
+/// \ingroup NTupleBrowse ROOT7
+/// \author Simon Leisibach <simon.satoshi.rene.leisibach@cern.ch>
+/// \date 2019-07-19
+/// \warning This is part of the ROOT 7 prototype! It will change without notice. It might trigger earthquakes. Feedback
+/// is welcome!
+
+/*************************************************************************
+ * Copyright (C) 1995-2019, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#include <algorithm>
+#include <cstring>
+#include <string>
+
+#include <Rtypes.h>
+
+#include <ROOT/RFieldVisitor.hxx>
+#include <ROOT/RNTupleBrowser.hxx>
+#include <ROOT/RPageStorageRoot.hxx>
+
+
+//--------------------------- RNTupleBrowser -----------------------------
+
+ClassImp(ROOT::Experimental::RNTupleBrowser);
+
+
+ROOT::Experimental::RNTupleBrowser::RNTupleBrowser(TDirectory* directory, int unittest): fDirectory{directory}, fReaderPtr{nullptr}, fUnitTest{unittest}, fCurrentTH1F{nullptr}
+{
+}
+
+
+void ROOT::Experimental::RNTupleBrowser::SetDirectory(TDirectory* directory)
+{
+   fDirectory = directory;
+   // Checks if a RNTupleReader with that directory was already created. If so it just takes the already existing RNTupleReader.
+   for(int i = 0; i < (int)fDirectoryVec.size(); ++i) {
+      if (fDirectoryVec.at(i) == fDirectory) {
+         fReaderPtr = fReaderPtrVec.at(i);
+         return;
+      }
+   }
+   
+   std::string fullPath = fDirectory->GetPath();
+   // The 5 lines below are used for unit tests in ntuplebrowse.cxx since a TDirectory can't be initialized with a path.
```

Comment:
Originally I wanted to do something like:
`TDirectory *fDirectory(filename, customPath)` or
`fDirectory->SetPath(customPath)`
but it's not possible. I'm currently trying to make it work with file->mkdir(...) but it's not working (but maybe later...)

Reply:
And I'm also trying to get rid of `fUnitTest`. The reason I created an extra member for the test is to allow communication between test and code to test. Because it's hard to test what a GUI (here TBrowser) did, I created means to check what code was executed. Which is what fUnitTest does, it checks if the proper functions were called.

Reply:
> but it's not possible. I'm currently trying to make it work with file->mkdir(...) 

I am a little confused ... the Browser is a view for the file and should not be modifying .. What are you trying to accomplish?

Reply:
The code would only execute in custom tests (unit test/gtest) where it creates a custom file/directory from scratch to see what it displays on the TBrowser. 
I managed to move that code from RNTupleBrowser.cxx to the test-file (test/ntuple_browse.cxx), so the issue is solved.